### DISCUSSION TITLE
Throw on invalid hooks type

### DISCUSSION THREAD

**Discussion by WebReflection** [pyscript/pyscript#1758] [Proposals]
TS (or better, JSDoc TS) is good and everything but it disappears once deployed as all we get at the end is just JS, and it has no guards around typed expectations.

This is nothing new, but especially for exposed APIs I think satisfying expectations is crucial, as well as throwing when such expectations aren't met, and we're also a bit obsessed with meaningful errors ... so hear me out:

How about we add this tiny module of mine that would grant that our hooks are actually used as expected and *fail instantly* if that's not the case?

https://github.com/WebReflection/type-checked-collections#readme

The main reason is that hooks can accept either callbacks (main thread) or just code (worker threads) and I can see already people trying to pass callbacks to worker hooks or mistakenly pass code to main thread hooks.

Any thoughts on this? The final size *should* :tm: not be increased much, as the module is little + it has a `/dummy` export escape whenever we want to debug something and be sure that code is not on our way.

/cc @antocuni @ntoll @JeffersGlass

**Comment by JeffersGlass** [comment]
As you say, this would allow us to raise an an error when you pass the wrong kind of _thing_ (code or callbacks) to hooks in either kind of thread, yes?

I'm +1 on that idea - and it looks like the total size of the _whole module_ is <20kb, which seems like a really decent tradeoff. I say go for it!

**Comment by WebReflection** [comment]
@antocuni @ntoll if neither of you is against this I'll move this discussion as issue and bring that module in.

**Reply by antocuni** [reply]
yes, it seems like a good idea to me 👍

**Reply by ntoll** [reply]
This is definitely a good idea. Doubly so because we have a drop in solution. I believe it will save folks hours of frustration and trying to debug code. :+1:
